linked-list.js

```diff
@@ -14,62 +14,188 @@ class LinkedList {
     this.head = null;
     this.tail = null;
     this.length = 0;
-
+    this.vals = vals
     for (let val of vals) this.push(val);
+
   }
 
   /** push(val): add new value to end of list. */
 
   push(val) {
-
+    let newNode = new Node(val)
+    if (this.head === null) {
+      this.head = newNode
+      this.tail = newNode
+    } else {
+      this.tail.next = newNode
```

Comment:
why would tail.next and tail be both newNode? shouldn't tail = newNode only? tail.next should always be null since tail is the last node